File: _data_for_notebooklm/ahtt_pdfs/pdf_merger.js
const fs = require('fs');
const path = require('path');
const { PDFDocument } = require('pdf-lib');

async function processPDFs() {
  const dir = './';
  // Get and sort PDF files A-Z
  const files = fs.readdirSync(dir)
    .filter(f => f.endsWith('.pdf'))
    .sort();

  // Group files by year (YYYY-...)
  const groups = {};
  files.forEach(f => {
    const year = f.split('-')[0];
    if (!groups[year]) groups[year] = [];
    groups[year].push(f);
  });

  for (const year in groups) {
    const mergedDoc = await PDFDocument.create();
    
    // Merge all files for this year
    for (const file of groups[year]) {
      const bytes = fs.readFileSync(path.join(dir, file));
      const doc = await PDFDocument.load(bytes);
      const copiedPages = await mergedDoc.copyPages(doc, doc.getPageIndices());
      copiedPages.forEach(page => mergedDoc.addPage(page));
    }

    const totalPages = mergedDoc.getPageCount();

    if (['2010', '2011', '2012', '2013', '2014'].includes(year) && totalPages > 1) {
      const mid = Math.floor(totalPages / 2);
      
      // Part 1
      const p1 = await PDFDocument.create();
      const p1Pages = await p1.copyPages(mergedDoc, Array.from({length: mid}, (_, i) => i));
      p1Pages.forEach(p => p1.addPage(p));
      fs.writeFileSync(`${year}_ahtt_01.pdf`, await p1.save());

      // Part 2
      const p2 = await PDFDocument.create();
      const p2Pages = await p2.copyPages(mergedDoc, Array.from({length: totalPages - mid}, (_, i) => i + mid));
      p2Pages.forEach(p => p2.addPage(p));
      fs.writeFileSync(`${year}_ahtt_02.pdf`, await p2.save());
      
      console.log(`Created ${year} Part 1 & 2`);
    } else {
      // Standard Merge
      fs.writeFileSync(`${year}_ahtt.pdf`, await mergedDoc.save());
      console.log(`Created ${year}_ahtt.pdf`);
    }
  }
}

processPDFs().catch(console.error);

File: _data_for_notebooklm/ahtt_pdfs/pdf_merger.py
import os
from pypdf import PdfWriter, PdfReader

# Get and sort files
files = sorted([f for f in os.listdir('.') if f.endswith('.pdf')])

# Group files by year (assumes format: YYYY-...)
years = {}
for f in files:
    year = f.split('-')[0]
    years.setdefault(year, []).append(f)

for year, fs in years.items():
    merger = PdfWriter()
    for f in fs:
        merger.append(f)
    
    # Logic for 2010, 2011, 2012
    if year in ['2010', '2011', '2012', '2013', '2014']:
        total_pages = len(merger.pages)
        mid = total_pages // 2
        
        # Part 1
        p1 = PdfWriter()
        for i in range(mid): p1.add_page(merger.pages[i])
        with open(f"{year}_01.pdf", "wb") as f1: p1.write(f1)
        
        # Part 2
        p2 = PdfWriter()
        for i in range(mid, total_pages): p2.add_page(merger.pages[i])
        with open(f"{year}_02.pdf", "wb") as f2: p2.write(f2)
    else:
        with open(f"{year}.pdf", "wb") as f_out:
            merger.write(f_out)
    
    merger.close()
